RNA-seq/03-gastric_cancer_exploratory.Rmd

```diff
@@ -166,10 +134,10 @@ ggplot2::ggsave(pca_plot_file, plot = ggplot2::last_plot())
 We don't have batch information (i.e., when the samples were run) for this 
 particular experiment, but let's imagine that `SRR585574` and `SRR585576` were
 run separately from all other samples.
-We'll put this in new "toy" column in `sample_meta_df`.
+We'll put this in new "toy" column in `gene_summarized`.
 
 ```{r batches}
-sample_meta_df$batch <- c("batch1", "batch1", "batch1", "batch1", "batch2", 
+gene_summarized@colData$batch <- c("batch1", "batch1", "batch1", "batch1", "batch2", 
                           "batch1", "batch2", "batch1")
```

Comment:
I think a better way to do this would be to do it in a few steps: start by pulling out the column data, then add the batch, then put it back in colData:
```suggestion
sample_info <- colData(gene_summarized)
sample_info$batch <- c("batch1", "batch1", "batch1", "batch1", "batch2", 
                          "batch1", "batch2", "batch1")
# replace the sample information                        
colData(gene_summarized) <- sample_info
```

One question I have is whether we need to actually do the next step of remaking the DESeq object. We could just put the colData into that object with the added batch column , and it would have the same effect (I think...). The only thing missing would be the comment about how you would "really" make the DESeq object if you had the batch information, but I am not sure that point comes across very well as it stands.

Reply:
Right. I think the idea is to show that you would need to remake the DESeq object because you *would* include batch in the equation *if* the batch info were real. But yeah, that message isn't super apparent.

Reply:
The reason we have to re-run this step is because of the `plotPCA()` step later that uses `batch` in the `intgroup` argument.

Reply:
Added some more explanation at this step. See how you feel about it.

Reply:
> The reason we have to re-run this step is because of the `plotPCA()` step later that uses `batch` in the `intgroup` argument.

Right, I get that, but if we did `colData(vst_data) <- sample_info` it would work just fine, I think.

Reply:
Oh yeah, that seems like it would work. Will try that.

Reply:
That worked. And I *think* it's not more confusing now. See what you think.